### DISCUSSION TITLE
Адреса размещения руководств IG

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#259] [General]
(от Андрея Забелина)

Сквозные ссылки, например из Lab IG в Core IG, будут работать в том случае, если Core IG будет добавлен в FHIR Package Registry (https://registry.fhir.org/). 

После добавления Core IG в FHIR Package registry можно будет обновить Lab IG конфигурацию sushi, указав зависимость от Core IG  (https://fshschool.org/docs/sushi/configuration/#dependencies)

Только после этого сквозные ссылки начнут работать

Также есть уточняющие вопросы по поводу того, где должны находиться IG. Имелось ли в виду, что по адресу fhir.ru/core должен находиться Core IG rucore.fhir.ru? Если да, то как быть с информацией, которая уже есть на странице fhir.ru/core? То же касается и fhir.ru/lab с rulab.fhir.ru.

Если контент по адресам fhir.ru/core и fhir.ru/lab можно проигнорировать и разместить там rucore.fhir.ru и rulab.fhir.ru, то это можно сделать

Если контент на этих страницах все-таки важен, то лучше всего будет использовать в качестве канонических ссылок rucore.fhir.ru и rulab.fhir.ru

**Comment by eugenekogan-spb** [comment]
@Vrodvi

**Reply by Vrodvi** [reply]
Есть сообщество FHIR-ru, для которого получилось выделить хост fhir.ru.

У сообщества есть два блока материалов:
- база знаний (Zendoc-сайт с информационными страницами)
- руководства по реализации FHIR (ig-сайты с информацией о руководствах, сформированные по рекомендациям FHIR (через SUSHI, publisher и все сопутствующие материалы)) 

Важны и нужны оба блока материалов, они не исключают друг друга.

url ресурсов, на мой взгляд, должен явно обозначать принадлежность к этим блокам.
Но сейчас url, наоборот, вносят путаницу, т.к. по url `http://fhir.ru/core` (указанному в качестве канонического для руководства) размещается не руководство, а соответствующая ему страница базы знаний.

Вариант разведения URL с явным укзаанием блока:
- раздел базы значений, посвящённый корневому руководству FHIR, следует разместить по адресу:
  - `https://fhir.ru/wiki/core` (вместо wiki может быть zendoc, knowledge-base или что-то аналогичное)
- само руководство RuCoreIG следует разместить по адресу:
  - `https://fhir.ru/ig/core`

(если учесть возможность продвижения версий руководства, то ещё заложить версию в url - `https://fhir.ru/wiki/core/v1` и `https://fhir.ru/ig/core/v1`)  
  
  
При этом также нужно каким-то образом учесть "канонические url" используемые в FHIR для идентификации артефактов, которые строятся унифицировано относительно некой базовой части url.
Эти канонические url строятся примерно-одинаково в различных зарубежных руководствах, к примеру:
```
в базовой FHIR спецификации: http://hl7.org/fhir/StructureDefinition/Device
в США:                       http://hl7.org/fhir/us/core/StructureDefinition/us-core-allergyintolerance
в Канаде:                    http://hl7.org/fhir/ca/baseline/StructureDefinition/profile-device-medical
в Европе:                    http://hl7.eu/fhir/ig/pcsp/StructureDefinition/Condition-otherinfos-eu-pcsp
```

Замечено, что во всех зарубежных руководствах канонические URL отличаются от фактических URL, по которым расположены сами артефакты (в частности, в фактических url учитывается версия артефакта, в то время как канонический url её не содержит).
Зачастую, при переходе по каноническому URL происходит перенаправление на фактический URL, что сглаживает негативный эффект с этим различием.

К примеру:
```
http://hl7.org/fhir/StructureDefinition/Device ( https://hl7.org/fhir/device.html )

http://hl7.org/fhir/ca/baseline/ImplementationGuide/hl7.fhir.ca.baseline ( https://build.fhir.org/ig/HL7-Canada/ca-baseline/index.html )
http://hl7.org/fhir/ca/baseline/StructureDefinition/profile-device-medical ( https://build.fhir.org/ig/HL7-Canada/ca-baseline/StructureDefinition-profile-device-medical.html )

http://hl7.org/fhir/us/core/ImplementationGuide/hl7.fhir.us.core ( https://hl7.org/fhir/us/core/STU6.1/index.html )
http://hl7.org/fhir/us/core/StructureDefinition/us-core-allergyintolerance ( https://hl7.org/fhir/us/core/STU6.1/StructureDefinition-us-core-allergyintolerance.html )

http://hl7.eu/fhir/ig/pcsp/ImplementationGuide/hl7.eu.fhir.pcsp ( https://build.fhir.org/ig/hl7-eu/pcsp/ ) 
http://hl7.eu/fhir/ig/pcsp/StructureDefinition/Condition-gvhd-eu-pcsp ( https://build.fhir.org/ig/hl7-eu/pcsp/StructureDefinition-Condition-gvhd-eu-pcsp.html )
```

**Reply by Vrodvi** [reply]
Похоже понадобится завести три набора URL:

- (1) фактические url для базы знаний (предположительно `https://fhir.ru/wiki`)
- (2.1) каноничекие url для руководств (предположительно `https://fhir.ru/ig`)
- (2.2) фактические url для руководств (предположительно `https://fhir.ru/ig/{руководство}/v{версия руководства}/{пути, сформированные используемым инструментарием SUSHI/Publisher}`)

Также желательно реализовать механизм перенаправления из канонических url в фактические для руководств, по аналогии с тем как это реализовано в базовой спецификации FHIR (к примеру, переход с канонического http://hl7.org/fhir/StructureDefinition/Patient) или в американском национальном руководстве (к примеру, переход с канонического http://hl7.org/fhir/us/core/ImplementationGuide/hl7.fhir.us.core))

**Reply by eugenekogan-spb** [reply]
Если уйти от перенаправления не удается, то возможно следует канонические url для руководств (2.1) не утяжелять добавлением "ig", а оставить как сейчас?

**Reply by Vrodvi** [reply]
У нас есть адекватные примеры руководств, все они имеют в построении url чётко прослеживаемую структуру.

В США и канаде структура url не содержит "ig" поскольку там вместо него выделен узел "fhir", затем код страны, затем код руководства `http://hl7.org/` `fhir/` `us/` `core`
Т.е. тут понятно по структуре что речь идёт про национальные руководства по реализации FHIR.

В европейском руководстве вообще явно выделен узел "ig" - `http://hl7.eu/fhir/ig/pcsp/`

У нас же что в этом `https://fhir.ru/core`, что в этом `https://rucore.fhir.ru` варианте не ясно что это вообще - руководства или просто страницы базы знаний. Тем более что по адресу `https://fhir.ru` вообще размещен zendoc.

**Reply by Vrodvi** [reply]
Если по зарубежным URL пройтись, то будет последовательно: 
https://hl7.org/fhir - сама спецификация FHIR
https://hl7.org/fhir/us - реестр руководств по реализации FHIR в США
https://hl7.org/fhir/us/core - одно конкретное руководство по реализации FHIR в США

![image](https://github.com/fhir-ru/core/assets/92928137/dd9fd53a-6845-423f-a656-8b93e6d30a9c)


В url fhir-ru пока-что очевидной логики построения путей не прослеживается.

В связи с этим я призываю к присвоению значащих url путей с чёткой сткрутурой, даже если она будет немного избыточной.

**Reply by Vrodvi** [reply]
В зарубежных руководствах, кстати, тоже есть ссылки на дополнительные сайты (базы знаний):

![image](https://github.com/fhir-ru/core/assets/92928137/bfb04ac2-38c7-494e-addc-147ab6112a76)

![image](https://github.com/fhir-ru/core/assets/92928137/22145204-e3ff-47e7-9a80-9b3f785aaf00)


Для руководств и для этих дополнительных сайтов предусмотрены отдельные явно обозначенные наборы URL, каждый со своей чёткой структурой.

**Reply by Vrodvi** [reply]
@eugenekogan-spb @OlegPenzin 

Нашёл пример (для Великобритании) организации иерархии сайтов, в том виде, как это я себе представлял:

![image](https://github.com/fhir-ru/core/assets/92928137/3d1a4668-c416-4030-a27b-612c9dc73616)

**Reply by Vrodvi** [reply]
И ещё важно не забыть про адрес размещения русскоязычного перевода самой спецификации FHIR, который также планировалось готовить когда-то в будущем.

Сейчас устаревшая версия перевода находится по адресу https://fhir-ru.github.io/index.html :
![image](https://github.com/fhir-ru/core/assets/92928137/3b1347de-7065-450a-b13b-c51f6718162e)

Итого, на данный момент с учётом всем обсуждённых моментов получается следующая приблизительная иерархия сайтов:
 
| адрес | описание |
|-----|-----|
|   |   |
|   | **Организационная часть** |
|  http://www.hl7.org |  сайт HL7  |
|  http://www.hl7.org.ru |   сайт российского филиала HL7  |
| https://fhir.ru  |  "лэндинговый сайт" FHIR в РФ  |
|   |   |
|   | **Перевод базовой спецификации** |
| https://fhir.ru/spec/  |  русский перевод базовой спецификации FHIR (последняя опубликованная версия) |
| https://fhir.ru/spec/r5  |  русский перевод базовой спецификации FHIR (5 версии) |
| https://fhir.ru/spec/...  |  русский перевод базовой спецификации FHIR (... версии) |
|   |   |
|   | **Формализованное описание руководств по реализации** |
| https://fhir.ru/ig | реестр российских руководств по реализации FHIR |
| https://fhir.ru/ig/core | формализованное представление (IG сайт) руководства по реализации "Ru Core" (последняя опубликованная версия) |
| https://fhir.ru/ig/core/v1 | формализованное представление (IG сайт) руководства по реализации "Ru Core" (1 версия) |
| https://fhir.ru/ig/core/... | формализованное представление (IG сайт) руководства по реализации "Ru Core" (... версия) |
| https://fhir.ru/ig/lab/v1 | формализованное представление (IG сайт) руководства по реализации "Ru Лаборатоная диагностика" (1 версия) |
| https://fhir.ru/ig/lab/... | формализованное представление (IG сайт) руководства по реализации "Ru Лаборатоная диагностика" (... версия) |
| https://fhir.ru/ig/... | формализованное представление (IG сайт) руководства по реализации "..." |
|   |   |
|   | **База знаний** |
| https://fhir.ru/wiki | База знаний FHIR в РФ |
| https://fhir.ru/wiki/core | Раздел базы знаний по руководству "Ru Core" (последняя опубликованная версия) |
| https://fhir.ru/wiki/core/v1 | Раздел базы знаний по руководству "Ru Core" (1 версия) |
| https://fhir.ru/wiki/lab | Раздел базы знаний по руководству "Ru Лаборатоная диагностика" (последняя опубликованная версия) |
| https://fhir.ru/wiki/lab/v1 | Раздел базы знаний по руководству "Ru Лаборатоная диагностика" (1 версия) |

EDIT:
Обсудили альтернативный вариант формирование базы знаний - база ведётся без версионирования, но при публикации формализованных руководств некоторые описания из базы знаний переносятся и фиксируются в конкретной опубликованной версии руководства.
В таком варианте необходимо обсудить и реализовать механизм разметки наполнения базы значений, которое должно переноситься в руководство, и сам процесс переноса (описательные страницы должны быть расположены иерархически в структуре руководства - нужно понимать как обозначить это расположение для переноса).

С учётом этого вараинта раздел с базой знаний может быть таким: 
| адрес | описание |
|-----|-----|
|   |   |
|   | **База знаний** |
| https://fhir.ru/wiki | База знаний FHIR в РФ |
| https://fhir.ru/wiki/core | Раздел базы знаний по руководству "Ru Core" (актуальная текущая версия) |
| https://fhir.ru/wiki/lab | Раздел базы знаний по руководству "Ru Лаборатоная диагностика" (актуальная текущая версия) |

**Comment by eugenekogan-spb** [comment]
Предложения Олега Пензина @OlegPenzin  по странице лэндинга:
Я бы видел его как дизайн отсюда https://dtxalliance.org/ совмещенный с контентом для привлечения новых участников как здесь https://darwin-eu.org/

**Reply by eugenekogan-spb** [reply]
В структуре https://dtxalliance.org/ я бы предложил такие куски текста:

Главный посыл: Развиваем российское здравоохранение с помощью международного стандарта обмена медицинскими данными HL7 FHIR (улучшаем / повышаем качество)

Сообщество объединяет компании, разрабатывающие программное обеспечение для российского здравоохранения, действует при поддержке ЦНИИОИЗ МЗ РФ. Целью сообщества является создание национальной модели данных RuCore (ссылка) и прикладных руководств (ссылка) для обмена данными в конкретных ситуациях.

Хочешь познакомиться с FHIR? Мы подготовили список вводных лекций (ссылка) про стандарт в целом и на конкретные темы, например профилирование, терминологии или SmartOnFHIR.

Хочешь участвовать в работе сообщества? Приходи в телеграмм-чат fhir-ru и на встречи онлайн по вторникам в 13:30 (https://us02web.zoom.us/j/9528778806)
